web/docs/auth/overview.md

```diff
@@ -2,26 +2,631 @@
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```wasp
+page MainPage {
+  component: import Main from "@client/pages/Main.jsx",
+  authRequired: true
+}
+```
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```wasp
+page MainPage {
+  component: import Main from "@client/pages/Main.tsx",
+  authRequired: true
+}
+```
+</TabItem>
+</Tabs>
+
+:::caution Requires auth method
+`authRequired` can only be used if some auth method is enabled for the app.
+:::
+
+If `authRequired` is set to `true`, the React component of the page (specified by `component` property) will be provided `user` object as a prop. Read more about the `user` object in the [Accessing the logged-in user](#accessing-the-logged-in-user) section.
+
+## Logout action
+
+We provide an action for logging out the user. Here's how you can use it:
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```jsx title="client/components/LogoutButton.jsx"
+import logout from '@wasp/auth/logout'
+
+const LogoutButton = () => {
+  return (
+    <button onClick={logout}>Logout</button>
+  )
+}
+```
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```tsx title="client/components/LogoutButton.tsx"
+import logout from '@wasp/auth/logout'
+
+const LogoutButton = () => {
+  return (
+    <button onClick={logout}>Logout</button>
+  )
+}
+```
+</TabItem>
+</Tabs>
+
+## Accessing the logged-in user
+
+You can get access to the `user` object both in the backend and on the frontend.
+
+### On the client
+
+#### Using the `user` prop
+
+If `authRequired` is set to `true` for a `page` declaration, the React component of the page will be provided `user` object as a prop:
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```wasp title="main.wasp"
+// ...
+
+page AccountPage {
+  component: import Account from "@client/pages/Account.jsx",
+  authRequired: true
+}
+```
+
+```jsx title="client/pages/Account.jsx"
+import Button from './Button';
+import logout from '@wasp/auth/logout';
+
+const AccountPage = ({ user }) => {
+  return (
+    <div>
+      <Button onClick={logout}>Logout</Button>
+      {JSON.stringify(user, null, 2)}
+    </div>
+  );
+};
+
+export default AccountPage;
+```
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```wasp title="main.wasp"
+// ...
+
+page AccountPage {
+  component: import Account from "@client/pages/Account.tsx",
+  authRequired: true
+}
+```
+
+```tsx title="client/pages/Account.tsx"
+import type { User } from '@wasp/entities';
+import Button from './Button';
+import logout from '@wasp/auth/logout';
+
+const AccountPage = ({ user }: { user: User }) => {
+  return (
+    <div>
+      <Button onClick={logout}>Logout</Button>
+      {JSON.stringify(user, null, 2)}
+    </div>
+  );
+};
+
+export default AccountPage;
+```
+</TabItem>
+</Tabs>
+
+#### Using the `useAuth` hook
```

Comment:
Can we explain when they should use the prop as opposed to the hook (and vice versa)?